### ISSUE TITLE
Wappalyzer prevents the Australian Taxation Office's business portal from working

### ISSUE THREAD

**Issue by F21** [AliasIO/Wappalyzer#2675]
**Do you want to request a *feature*, a *new application detection* or report a *bug*?**
*bug*

**Is your issue about WebExtension driver (Chrome & Firefox), the website, the NPM driver or the bookmarklet ?**
Chrome Extension

**What is the current behavior ?**
When the Wappalyzer extension is enabled, I am not able to open the ATO's business portal. The business portal will fail with the error message: `An error has occurred while detecting your system configuration or accessing the required software on your machine.`. If I disable the Wappalyzer extension, it works correctly. It's possible that this could be something to do with the business portal and not Wappalyzer.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a demo of the problem.**
Visit https://bp.ato.gov.au/ with the Wappalyzer extension enabled.
Click `Login`. 
A popup should open and it should check your software configuration.
After waiting for a long time, you will receive the following error message: `An error has occurred while detecting your system configuration or accessing the required software on your machine.`

**What is the expected behavior ?**
The business portal should load correctly.

**Which versions of Wappalyzer, and which browser / OS are affected by this issue ? Did this work in previous versions ?**
Wappalyzer: 5.7.4
Chrome:  73.0.3683.103
Windows 10 64-bit 1809 17763.437

**Comment by kyletaylored** [comment]
I have Wappalyzer enabled and I'm able to see this screen. Is this what you should see after the software check?

![image](https://user-images.githubusercontent.com/1759794/56431576-cbc45d80-628f-11e9-8a16-6aacac1cea6c.png)

**Comment by F21** [comment]
I am not sure why you're directed to the myGov sign in page directly. There are currently 2 methods to authenticate for the business portal:
- An AUSKey, which uses PKI. In this case, a chrome extension is needed and it should show a list of keys to log in with.
- A MyGov account to authenticate using a username and password.

If MyGov is selected, then there is the option to force it to always use MyGov for logging in. I've found that it's possible to turn off Wappalyzer and ask it to always use MyGov. Since the business portal will always redirect to MyGov for logging in, turning Wappalyzer no longer breaks it. However, if one wants to login using an AUSKey, then it's not possible with the Wappalyzer extension enabled.

I think you should be able to reproduce the problem if you use incognito mode, or clear cookies for the ato.gov.au domain and its subdomains.

Screenshot of the error:
![err](https://user-images.githubusercontent.com/2263040/56450068-59be4800-6365-11e9-9ad7-24f96ad0bfd1.png)

Screenshot of what's expected:
![success](https://user-images.githubusercontent.com/2263040/56450079-6b9feb00-6365-11e9-8beb-32f723fb5f72.png)

**Comment by kyletaylored** [comment]
@F21 So I've gotten the screenshot you're saying is correct almost every time, with Wappalyzer enabled.

But looking at their docs for AUSKey that says you have to install their own Chrome extension (and seeing that their own site throws errors), I would say it's all bets are off to claim it's a Wappalyzer issue.

**Errors thrown in their popup**
![image](https://user-images.githubusercontent.com/1759794/56483071-25de4200-648d-11e9-831d-0ae6488d4a5d.png)

Could you turn on developer mode and watch for errors in the console to see what happens when the AUSKey extension is installed?

**Comment by F21** [comment]
The AUSKey extension does not have any effect on whether it works or not. It just shows the list of AUSKeys available in the dropdown. If I don't have it installed, with Wappalyzer turned off, it loads successfully, but I just don't see they keys in the dropdown.

I am able to confirm this consistently. Whenever Wappalyzer is enabled (this is using incognito mode with all extensions disabled, except for Wappalyzer), the system error message comes up. If I turn off Wappalyzer, it loads correctly.

The same is the case without using incognito more (with a bunch of other extensions enabled). If Wappalyzer is off, it loads correctly. If Wappalyzer is on, it fails.

Regardless of whether Wappalyzer is on or off, I see the same logs in the console. The missing font files don't seem to be a problem as they show up when Wappalyzer is off and the page loads correctly:
![Untitled](https://user-images.githubusercontent.com/2263040/56483399-1bfc2380-650d-11e9-9cf7-1cf8a72b8f8f.png)

I am going to install firefox and try the firefox extension to see if it's reproducible there.

**Comment by F21** [comment]
Just did the following tests:

Windows 8.1 64-bit VM with Chrome 73.0.3683.103 (Official Build) (64-bit): Page fails with Wappalyzer enabled. If Wappalyzer is disabled it loads correctly.

Windows 10 64-bit (latest) VM with Firefox 66.0.3 (64-bit): The page works whether Wappalyzer is on or off.

I think this is most likely due to a problem with the Chrome extension as I can reproduce it in Chrome on 2 different OSes.

**Comment by kyletaylored** [comment]
Alright, even though I don't have an authorized key, I went ahead and installed the AUSKey package / Chrome extension my macOS (though, I don't actually see it anywhere). My first reaction is after it made me log out and back in, all my Chrome extensions were missing except one until I went to my Manage Extensions screen and checked one on/off, then they reloaded.

There's also two console windows - one for the normal browser, and one for the extension as a debugger. When I open the extension console, I can see this failing Promise for the Browser Polyfill (which ironically enough is maintained by [Mozilla](https://github.com/mozilla/webextension-polyfill)). I can see Wappalyzer is on the latest version, 0.4.0.

![image](https://user-images.githubusercontent.com/1759794/56485126-67281f00-6498-11e9-835a-82100af5cb2a.png)

That seems to be the underlying issue here, and is probably why it works in Firefox since it's native. I might take a look later, but that seems to be the issue right now.
